docs/sphinx/BUILD.bazel

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@docs_deps//:requirements.bzl", "requirement")
+load("@rules_python//python:pip.bzl", "compile_pip_requirements")
+load("@rules_python//python:py_binary.bzl", "py_binary")
+load(":docgen.bzl", "sphinx_docs", "sphinx_stardocs")
+
+# See README.md for instructions. Short version:
+# * `bazel run //docs/sphinx:docs.serve` in a separate terminal
+# * `ibazel build //docs/sphinx:docs` to automatically rebuild docs
+sphinx_docs(
+    name = "docs",
+    srcs = [
+        ":stardocs",
+    ] + glob(
+        include = ["*.md"],
+    ),
+    config = "conf.py",
+    formats = [
+        "html",
+        "latex",
+    ],
+    sphinx = ":sphinx_build",
+    strip_prefix = package_name() + "/",
+)
+
+sphinx_stardocs(
+    name = "stardocs",
+    docs = {
+        "api/cc/py_cc_toolchain_info.md": "//python/cc:py_cc_toolchain_info_bzl",
+        "api/defs.md": "//python:defs_bzl",
+        "api/entry_points/py_console_script_binary.md": "//python/entry_points:py_console_script_binary_bzl",
+        "api/packaging.md": "//python:packaging_bzl",
+        "api/pip.md": "//python:pip_bzl",
+        "api/cc/py_cc_toolchain.md": dict(
+            dep = "//python/private:py_cc_toolchain_bzl",
+            input = "//python/private:py_cc_toolchain_rule.bzl",
+        ),
+    },
+    tags = ["docs"],
+)
+
+py_binary(
+    name = "sphinx_build",
+    srcs = ["sphinx_build.py"],
+    deps = [
+        requirement("sphinx"),
+        requirement("sphinx_rtd_theme"),
+        requirement("myst_parser"),
+        requirement("readthedocs_sphinx_ext"),
+    ],
+)
+
+# Run bazel run //docs/sphinx:requirements.update
+compile_pip_requirements(
+    name = "requirements",
+    requirements_in = "requirements.in",
+    requirements_txt = "requirements.txt",
+    # The requirements output differs on Windows, so just restrict it to Linux.
+    # The build process is only run on, and only works for, Linux anyways.
```

Comment:
nit: what are the reasons to not support Mac?

Reply:
Mostly because the docs will only be built on Ubuntu. Mac should be similar enough it shouldn't be an issue; added requirements_darwin